src/DetectorConstruction.cpp

```diff
@@ -18,18 +18,22 @@
 #include "G4SDManager.hh"
 #include "G4LogicalVolumeStore.hh"
 
-
-DetectorConstruction::DetectorConstruction(): G4VUserDetectorConstruction()
-{}
+DetectorConstruction::DetectorConstruction(): G4VUserDetectorConstruction(), detectorConfiguration(true)
+{
+  fMessenger = new G4GenericMessenger(this, "/Geometry/", "Geometry configuration");
+  fMessenger->DeclareProperty("DetectorConfiguration", detectorConfiguration, "True if HD, false if VD");
+}
```

Comment:
```suggestion
DetectorConstruction::DetectorConstruction(const bool& useHorizontalDrift): G4VUserDetectorConstruction(), detectorConfiguration(useHorizontalDrift)
{
  fMessenger = new G4GenericMessenger(this, "/Geometry/", "Geometry configuration");
  fMessenger->DeclareProperty("DetectorConfiguration", detectorConfiguration, "True if HD, false if VD");
}
```

Reply:
The value of detectorConfiguration is initialized to true by default and afterwards read from the .mac file. I don't see why we should pass `const bool& useHorizontalDrift=True` as an argument.